backend/src/utils/auth.ts

```diff
@@ -34,32 +35,37 @@ export interface CustomRequest extends Request {
 }
 
 export const checkAuth = async (req: Request, res: Response, next: NextFunction) => {
+    logger.debug("Auth called: checkAuth")
     try {
         const token = req.header('Authorization')?.replace('Bearer ', '');
         if (!token) {
+            logger.debug("Auth called: checkAuth error | No token found")
             throw new Error('No token found');
         }
 
         const decoded = jwt.verify(token, publicJwtKey);
         (req as CustomRequest).token = decoded;
         next();
     } catch (err) {
+        logger.debug(`Auth called: checkAuth error | ${err}`)
         res.status(401).send('Please authenticate');
     }
 };
 
 export function generateAsyncToken(payload: { id: string, accountName: string }): Promise<string> {
+    logger.debug(`Auth called: generateAsyncToken`)
     return new Promise((resolve, reject) => {
 
         if (!privateJwtKey) {
+            logger.debug(`Auth called: generateAsyncToken error | JWT PRIVATE KEY rejected`)
             reject(new Error("JWT PRIVATE KEY rejected."));
             return;
         }
 
         jwt.sign(payload, privateJwtKey, { algorithm: 'RS256' }, (err, token) => {
             if (err) {
                 reject(err);
-                console.error('Error generating token:', err);
+                logger.debug(`Auth called: generateAsyncToken error | Error generating token | ${err}`)
```

Comment:
If debug log level is not enabled this message will be handled anyways, consuming computer cicles. Please, check if you could invoke the message with placeholders like 
```
logger.debug("Auth called: generateAsyncToken error | Error generating token | $1", err)
```